source-academy/frontend
# Game location updates

### Description

- Adds configurable previews for locations
- Fixes `add_item` action for location items like talkTopics and navigation which may not have associated subscribers

### Type of change


### How to test

 - Use a preview header in the checkpoint file like in the example below, and make sure the correct nav preview appears
```
room
    modes: move, talk
    nav: room2, room3
    preview: /locations/yourRoom/emergency.png
```
- Try using the add_item action for location items like in this example:
```
enter
    @narrator
    Grandmaster Hartin Menz goes back to his seat.
        remove_item(talkTopics, room, HMDialogue5)
        add_item(talkTopics, room, HMDialogue5)
        add_item(navigation, room, room3)
        remove_item(navigation, room, room2)
```

### Checklist
